File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/test/TestImperativeEntity.java
package io.quarkus.it.mongodb.panache.test;

import io.quarkus.mongodb.panache.PanacheMongoEntity;

public class TestImperativeEntity extends PanacheMongoEntity {
    public String title;
    public String category;
    public String description;
    public int cpt = 1;

    public TestImperativeEntity() {
    }

    public TestImperativeEntity(String title, String category, String description) {
        this.title = title;
        this.category = category;
        this.description = description;
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/test/TestImperativeEntity.kt
package io.quarkus.it.mongodb.panache.test

import io.quarkus.mongodb.panache.kotlin.PanacheMongoCompanion
import io.quarkus.mongodb.panache.kotlin.PanacheMongoEntity

class TestImperativeEntity : PanacheMongoEntity {
    companion object : PanacheMongoCompanion<TestImperativeEntity>

    lateinit var title: String
    lateinit var category: String
    lateinit var description: String
    var ctp = 1

    constructor() {}

    constructor(title: String, category: String, description: String) {
        this.title = title
        this.category = category
        this.description = description
    }
}
